src/Application/ConfigResolver.php

```diff
@@ -90,13 +87,12 @@ public static function resolvePath(InputInterface $input): string
      */
     public static function guess(Composer $composer): string
     {
-        foreach (self::PRESETS as $presetClass) {
-            if ($presetClass::shouldBeApplied($composer)) {
-                return $presetClass::getName();
-            }
+        $preset = self::guessPresetClass('', $composer);
+        if ($preset !== '') {
+            return $preset::getName();
         }
 
-        return self::DEFAULT_PRESET;
+        return 'default';
```

Comment:
I think this function is not used anymore (except in test). Can you drop it and adjust tests ?